actors/builtin/miner/deadline_state.go

```diff
@@ -1076,6 +1204,79 @@ func (dl *Deadline) RescheduleSectorExpirations(
 	return allReplaced, nil
 }
 
+// DisputeInfo includes all the information necessary to dispute a post to the
+// given partitions.
+type DisputeInfo struct {
+	AllSectorNos, IgnoredSectorNos bitfield.BitField
+	DisputedSectors                PartitionSectorMap
+	DisputedPower                  PowerPair
+}
+
+// LoadPartitionsForDispute
+func (dl *Deadline) LoadPartitionsForDispute(store adt.Store, partitions bitfield.BitField) (*DisputeInfo, error) {
+	partitionsSnapshot, err := dl.PartitionsSnapshotArray(store)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to load partitions: %w", err)
+	}
+
+	var allSectors, allIgnored []bitfield.BitField
+	disputedSectors := make(PartitionSectorMap)
+	disputedPower := NewPowerPairZero()
+	err = partitions.ForEach(func(partIdx uint64) error {
+		var partitionSnapshot Partition
+		if found, err := partitionsSnapshot.Get(partIdx, &partitionSnapshot); err != nil {
+			return err
+		} else if !found {
+			return xerrors.Errorf("failed to find partition %d", partIdx)
+		}
+
+		// Record sectors for proof verification
+		allSectors = append(allSectors, partitionSnapshot.Sectors)
+		allIgnored = append(allIgnored, partitionSnapshot.Faults)
+		allIgnored = append(allIgnored, partitionSnapshot.Terminated)
+		allIgnored = append(allIgnored, partitionSnapshot.Unproven)
+
+		// Record active sectors for marking faults.
+		active, err := partitionSnapshot.ActiveSectors()
+		if err != nil {
+			return err
+		}
+		err = disputedSectors.Add(partIdx, active)
+		if err != nil {
+			return err
+		}
+
+		// Record disputed power for penalties.
+		//
+		// NOTE: This also includes power that was
+		// activated at the end of the last challenge
+		// window, and power from sectors that have since
+		// expired.
```

Comment:
Where? I could put it on top of the `ParitionsSnapshot` field, but it seems kind of obvious that the partition snapshot includes power that's no longer active. I put it here because the fact that we're "disputing" no longer active power is interesting.